rust_snuba/rust_arroyo/src/backends/kafka/types.rs

```diff
@@ -1,8 +1,9 @@
 use rdkafka::message::OwnedHeaders;
+use std::sync::Arc;
 
 #[derive(Clone, Debug)]
 pub struct KafkaPayload {
-    pub key: Option<Vec<u8>>,
-    pub headers: Option<OwnedHeaders>,
-    pub payload: Option<Vec<u8>>,
+    pub key: Option<Arc<Vec<u8>>>,
```

Comment:
```suggestion
    pub key: Option<Arc<[u8]>>,
```
That would avoid one indirection